src/managers/GuildInviteManager.js

```diff
@@ -0,0 +1,183 @@
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite without checking cache
+   * guild.invites.fetch({ code: 'DrzKVU3', force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single invite without caching
+   * guild.invites.fetch({ code: 'DrzKVU3', cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    const code = DataResolver.resolveInviteCode(options);
+    if (code) return this._fetchSingle({ code, cache: true });
+    if (!options.code) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+    }
+    return this._fetchSingle({
+      ...options,
+      code: DataResolver.resolveInviteCode(options.code),
+    });
+  }
+
+  async _fetchSingle({ code, cache, force = false }) {
+    if (!force) {
+      const existing = this.cache.get(code);
+      if (existing) return existing;
+    }
+
+    const invites = await this._fetchMany(cache);
+    const invite = invites.get(code);
+    if (!invite) throw new Error('INVITE_NOT_FOUND');
+    return invite;
+  }
+
+  async _fetchMany(cache) {
+    const data = await this.client.api.guilds(this.guild.id).invites.get();
+    return data.reduce((col, invite) => col.set(invite.code, this.add(invite, cache)), new Collection());
+  }
+
+  /**
+   * @typedef {CreateInviteOptions} CreateChannelInviteOptions
+   * @property {GuildChannel} [channel] The optional channel to define where to create the invite.
+   */
+
+  /**
+   * Create a invite to the guild from the provided or the first available channel.
+   * @param {CreateChannelInviteOptions} [options={}] The options for creating the invite from a optional channel.
+   * @returns {Promise<Invite?>}
+   * @example
+   * // Create an invite to a random allowed channel
+   * guild.invites.create()
+   *   .then(invite => console.log(`Created an invite with a code of ${invite.code} from ${invite.channel}`))
+   *   .catch(console.error);
+   * @example
+   * // Create an invite to a selected channel
+   * guild.invites.create({ channel: '599942732013764608' })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  create(options = {}) {
+    let channel;
+
+    if (!options.channel) {
+      channel = this.guild.channels.cache.find(c => c.permissionsFor(this.guild.me).has('CREATE_INSTANT_INVITE'));
+      if (!channel) return Promise.reject(new Error('GUILD_CHANNEL_INVITE'));
+    }
+
+    channel = this.guild.channels.resolve(options.channel);
+    if (!channel) return Promise.reject(new Error('GUILD_CHANNEL_RESOLVE'));
+
+    return channel.createInvite(options);
+  }
+
+  /**
+   * Deletes a invite.
+   * @param {InviteResolvable} invite The invite to delete
+   * @param {string} [reason] Reason for deleting the invite
+   * @returns {Promise<Invite>}
```

Comment:
Wrong return type

Reply:
Your right, I think that a void Promise will be better!